### PR TITLE
Match system.c (2 non matching)

### PR DESCRIPTION
**Pull request by Yanis42**
the remaining non-matching is a regalloc issue in systemExecute

### CODE REVIEW

File: src/emulator/system.c [hunk 1]
```diff
@@ -0,0 +1,1801 @@
...
+#include "macros.h"
+#include "revolution/os.h"
+#include "revolution/vi.h"
+#include "stdlib.h"
+#include "string.h"
+
+//! TODO: document
+extern _XL_OBJECTTYPE gClassHelpMenu;
+bool fn_8007D688(Rsp* pRSP, void** pBuffer, s32 unk1, s32 unk2);
+int fn_8005329C(Frame*, s32, s32, s32);
```
Comment (cadmic) [inline src/emulator/system.c:35]: move to proper headers? I think they all should be clear in this case

File: src/emulator/system.c [hunk 2]
```diff
@@ -0,0 +1,1801 @@
...
+    },
+    {
+        SOT_NONE,
+        NULL,
+        0,
+        {0},
+    },
+};
+
+#include "lbl_8016FEA0.inc"
```
Comment (cadmic) [inline src/emulator/system.c:288]: just curious, what is this? maybe add a short comment?

File: src/emulator/system.c [hunk 3]
```diff
@@ -0,0 +1,1801 @@
...
+
+static SystemRomConfig gSystemRomConfigurationList;
+
+bool systemSetStorageDevice(System* pSystem, SystemObjectType eStorageDevice, void* pArgument) {
+    switch (eStorageDevice) {
+        case SOT_EEPROM:
```
Comment (cadmic) [inline src/emulator/system.c:294]: merge/rebase?

File: src/emulator/system.c [hunk 4]
```diff
@@ -0,0 +1,1801 @@
...
+        return false;
+    }
+
+    nCode = 0;
+    for (iCode = 0; iCode < ARRAY_COUNT(anCode); iCode++) {
+        nCode += anCode[iCode];
+    }
+
+    // Ocarina of Time or Majora's Mask
+    if (pSystem->eTypeROM == 'NZSJ' || pSystem->eTypeROM == 'NZSE' || pSystem->eTypeROM == 'NZSP') {
```
Comment (cadmic) [inline src/emulator/system.c:423]: rom enums? ditto below

File: src/emulator/system.c [hunk 5]
```diff
@@ -0,0 +1,1801 @@
...
+    } else {
+        nSizeRAM = 0x400000;
+        nSizeCacheROM = 0x800000;
+    }
+
+    if (!ramSetSize(SYSTEM_RAM(gpSystem), nSizeRAM)) {
+        return false;
+    }
+
+    return !!ramWipe(SYSTEM_RAM(gpSystem));
```
Comment (cadmic) [inline src/emulator/system.c:446]: can this be written as `if (!ramWipe) ...`?

File: src/emulator/system.c [hunk 6]
```diff
@@ -0,0 +1,1801 @@
...
+            }
+            BUFFER_GET(pBuffer2, 4) = 0x17D6;
+            break;
+        case 'NRXP':
+        case 'NRXJ':
+        case 'NRXE':
+            if (!ramGetBuffer(SYSTEM_RAM(gpSystem), &pBuffer2, 0x300, NULL)) {
+                return false;
+            }
+            BUFFER_GET(pBuffer2, 4) = 0x17D6;
```
Comment (cadmic) [inline src/emulator/system.c:1055]: I find it a bit weird that BUFFER_GET() is used for setting too. Does it work if `pBuffer` is an s32* and we just do `pBuffer2[4]` etc here?

File: src/emulator/system.c [hunk 7]
```diff
@@ -0,0 +1,1801 @@
...
+    }
+
+    if (nMaskFinal != 0) {
+        if (!cpuException(SYSTEM_CPU(gpSystem), CEC_INTERRUPT, nMaskFinal)) {
+            return false;
+        }
+    } else {
+        if ((eCodeFinal != CEC_NONE) && !cpuException(SYSTEM_CPU(gpSystem), eCodeFinal, 0)) {
+            return false;
+        }
```
Comment (cadmic) [inline src/emulator/system.c:1687]: nit: split into two ifs probably

File: src/emulator/system.c [hunk 8]
```diff
@@ -0,0 +1,1801 @@
...
+                if (!systemGetException(pSystem, iException, &exception)) {
+                    return false;
+                }
+
+                bUsed = false;
+
+                if (exception.eCode == CEC_INTERRUPT) {
+                    if (cpuTestInterrupt(SYSTEM_CPU(gpSystem), exception.nMask) &&
+                        ((exception.eTypeMips == MIT_NONE) ||
+                         miSetInterrupt(SYSTEM_MI(gpSystem), exception.eTypeMips))) {
```
Comment (cadmic) [inline src/emulator/system.c:1660]: nit: redundant parens (in this case I think it makes the condition a bit harder to read, the important parens are the ones around the (x || y)

**Review by cadmic — COMMENTED** [review]
gave up on matching systemExecute :(

**Comment by Yanis42** [comment]
this regalloc is too strong

File: src/emulator/system.c (hunk 2 above)
Comment (Yanis42) [reply src/emulator/system.c:288]: it's hacks used by diddy kong racing, for some reasons it breaks systemSetupGameALL to use this symbol directly so for now I'm using the address and I block the reloc in the config

Comment (Yanis42) [reply src/emulator/system.c:288]: nvm looks like it's also used by paper mario, I need to dig into this to figure out if both usage are real lol, I'm a bit confused by this rught now

File: src/emulator/system.c (hunk 6 above)
Comment (Yanis42) [reply src/emulator/system.c:1055]: it worked for both buffers

**Review by cadmic — APPROVED** [review]
Guess we should consider systemSetupGameALL non-matching too because of the hardcoded data address

**Pull request merged**